src/potions.ts

```diff
@@ -79,6 +81,26 @@ for (const effectGroup of mutuallyExclusiveList) {
   }
 }
 
+const allNoHookahEffects = Effect.all().filter((effect) => !effect.attributes.includes("nohookah"));
+const validWishString: Map<Effect, string> = new Map();
+function getValidWishString(effect: Effect): string | undefined {
+  if (!validWishString.has(effect)) {
+    const allOtherNoHookahEffectNames = allNoHookahEffects
+      .filter((e) => e !== effect)
+      .map((e) => e.name.toLowerCase());
+    const validSubstrings = Array.from(
+      effect.name.toLowerCase().match(RegExp(/[\w ]+/g)) ?? []
+    ).filter((s) => !allOtherNoHookahEffectNames.some((name) => name.includes(s)));
+    validWishString.set(effect, validSubstrings[0]?.trim());
+  }
+  return validWishString.get(effect);
+}
+const validWishes = allNoHookahEffects
+  .filter((effect) => getValidWishString(effect))
+  .filter(
+    (effect) => (globalOptions.prefs.yachtzeechain ? effect !== $effect`Eau d' Clochard` : true) // hardcoded heuristics
+  );
```

Comment:
These can/should probably be merged

Reply:
```suggestion
  .filter((effect) => !!getValidWishString(effect) && (!globalOptions.prefs.yachtzeechain || effect !== $effect`Eau d' Clochard`));
```

you may or may not need the double bang here